internal/services/pam/pam.go

```diff
@@ -138,6 +138,15 @@ func (s Service) SelectBroker(ctx context.Context, req *authd.SBRequest) (resp *
 		lang = "C"
 	}
 
+	userIsDisabled, err := s.userManager.IsUserDisabled(username)
+	if err != nil && !errors.Is(err, users.NoDataFoundError{}) {
+		return nil, fmt.Errorf("could not check if user %q is disabled: %w", username, err)
+	}
+	// Throw an error if the user trying to authenticate already exists in the database and is disabled
+	if err == nil && userIsDisabled {
+		return nil, status.Error(codes.PermissionDenied, fmt.Sprintf("user %s is disabled", username))
+	}
```

Comment:
This check can't be moved at lower level in the stack so that it's not only PAM-related?

Reply:
I was thinking to have this in the brokers side, but as per your NSS comment it's not something we want then.